apps/src/studio/utils/commandQueue.js

```diff
@@ -0,0 +1,39 @@
+import { singleton as studioApp } from '@cdo/apps/StudioApp';
+import api from '../api';
+
+/**
+ * Execute the code for all of the event handlers that match an event name
+ * @param {string} name Name of the handler we want to call
+ * @param {boolean} allowQueueExtension When true, we allow additional cmds to
+ *  be appended to the queue
+ * @param {Array} extraArgs Additional arguments passed into the virtual JS
+ *  machine for consumption by the student's event-handling code.
+ */
+export function callHandler(name, allowQueueExtension, extraArgs = []) {
+  Studio.eventHandlers.forEach(function (handler) {
+    if (studioApp().isUsingBlockly()) {
```

Comment:
It would also be great to see fewer side effects on globals defined in `Studio`. Perhaps `Studio.eventHandlers` could be refactored into a class which handles all the various event operations on studio entirely within itself, and we could then simply have blockly and droplet versions of that class, determining at config time which one to instantiate?